packages/compass-editor/src/types.d.ts

```diff
@@ -0,0 +1,58 @@
+import type AceBuilds from 'ace-builds';
+import type { Ace } from 'ace-builds';
+
+type GlobalAce = {
+  EditSession: typeof AceBuilds['EditSession'];
+  Range: typeof AceBuilds['Range'];
+  UndoManager: typeof AceBuilds['UndoManager'];
+  VirtualRenderer: typeof AceBuilds['VirtualRenderer'];
+  config: typeof AceBuilds['config'];
+  createEditSession: typeof AceBuilds['createEditSession'];
+  edit: typeof AceBuilds['edit'];
+  require: typeof AceBuilds['require'];
+  version: typeof AceBuilds['version'];
+  acequire: typeof AceBuilds['require'];
+  define: (
+    module: string,
+    depdendencies: string[],
+    moduleFn: (
+      acequire: typeof AceBuilds['require'],
+      exports: any,
+      module: any
+    ) => void
+  ) => void;
+};
+
+declare global {
+  declare const ace: GlobalAce;
+}
+
+export interface HighlightRules {
+  normalizeRules(): void;
+  createKeywordMapper(
+    map: Record<string, string>,
+    defaultToken: string
+  ): (...args: any[]) => any;
+  $rules: Record<
+    string,
+    {
+      regex: string | RegExp;
+      token: string | string[] | ((...args: any[]) => any);
+      next?: string;
+    }[]
+  >;
+}
+
+export interface WorkerClient {
+  attachToDocument(doc: Document): void;
+}
+
+export interface AceMode {
+  HighlightRules: new (...args: unknown) => HighlightRules;
+  createWorker(session: Ace.EditSession): WorkerClient | null;
+}
+
+export type MongoDBCompletion = Ace.Completion & {
```

Comment:
`MongoDBCompletion` is not very self explanatory, what about `CompletionWithServerInfo` or something? .. sorry if i missed the point here, these `version` / `projectVersion` don't tell much about what they are .. maybe we have the opportunity to clarify it?

Reply:
Yeah, let's name it that. I couldn't really come up with a good name and like your suggestion. Version is supported server version, what is project version I actually have no idea 😆 I'll poke around and will try to add some meaningful description for them

Reply:
Done in https://github.com/mongodb-js/compass/pull/3589/commits/562c7b1287336e64d71281a249dc970e097eee28

Added a comment for `projectVersion`, but tbh we might just drop it down the road? I don't see any mention in the docs about some expressions not being supported by $project stage and the server version is way behind of what we dupport

Reply:
Yeah, I'm trying to recall why we may have done that 🤔 .. I feel i heard something in the past, but buh, maybe we need to investigate, good for another PR!

Reply:
Do you think we could rename `version` to `serverVersion` instead?

Reply:
Hmmm, we can, but this will be a change for this new mongodb-constants package in the shared monorepo first and then mongosh, vscode, and compass will need to be adjusted for it. Just want to double-check whether you think it would still be worth it in that case 🤔